packages/jest-snapshot/src/utils.ts

```diff
@@ -136,6 +136,21 @@ export const removeExtraLineBreaks = (string: string): string =>
     ? string.slice(1, -1)
     : string;
 
+export const removeLinesBeforeExternalMatcherTrap = (stack: string): string => {
+  const lines = stack.split('\n');
+
+  for (let i = 0; i < lines.length; i += 1) {
+    // It's a function name specified in `packages/expect/src/index.ts`
+    // for external custom matchers.
+    if (lines[i].includes('__EXTERNAL_MATCHER_TRAP__')) {
+      lines.splice(1, i);
```

Comment:
(connected to the change below)

the name of this function suggests to me it's meant to remove all lines before the thing it finds. If so, this is cleaner to me than mutating the array

```suggestion
      return `lines.slice(i).join('\n')
```

and ditch the `break`

If that alters the function to _not_ do what you want, I suggest we rename it

Reply:
Makes sense!